federation/pkg/federation-controller/secret/secret_controller.go

```diff
@@ -55,34 +55,37 @@ var (
 	RequiredResources = []schema.GroupVersionResource{apiv1.SchemeGroupVersion.WithResource("secrets")}
 )
 
-type SecretController struct {
-	// For triggering single secret reconciliation. This is used when there is an
-	// add/update/delete operation on a secret in either federated API server or
-	// in some member of the federation.
-	secretDeliverer *util.DelayingDeliverer
-
-	// For triggering all secrets reconciliation. This is used when
-	// a new cluster becomes available.
+// FederationSyncController synchronizes the state of a federated type
+// to clusters that are members of the federation.
+type FederationSyncController struct {
```

Comment:
cool, as I said, I think its still great the way it is as now. :)
and I think bot needs both approve and lgtm.
/lgtm